### PR TITLE
Bootstrap code and most overlays, and hopefully fix progress script

### PR DESCRIPTION
**Pull request by AngheloAlf**
- Add every overlay which already has a C file to the spec instead of the asm files. Overlays missing C files:
  - `ovl_title`.
  - gamestates
  - `ovl_daytelop` 
  - kaleido
  - `ovl_Player_Actor`. Its C file already is in a [PR in MM](https://github.com/zeldaret/mm/pull/212), so I wouldn't worry too much about it.
  - `ovl_En_Fishing`, which is already in master.
  - effects
  - `ovl_fbdemo_*`
- Made the missing C files for `code` and add most of them to the spec. This are the files that are still using the asm directly in the spec:
  - ~~`z_scene` is decomped but including it to the spec would not OK. I suspect the `z_scene.data` split is wrong but I haven't checked.~~ Edit: It had a few data splits problem, but after a few tries I managed to OK it.
  - ~~`z_skelanime` which is decomped too. I haven't looked too much what the issue is with that one :s~~. Edit: I forgot to remove its bss when I tried it, now it is OK.
  - `code_80185F90` which seems to be handwritten. I guess there isn't much we can do here.
  - `code_801A51F0` (not decomped), its first function has a jump table that is bigger than the function itself, and the assembler will complain about it. I also left a comment in the C file about this.
- Add the extracted scenes to the spec.
- Re-enable C files compilation in Makefile.
- Updated the already existing `code` C files to use the new assembly paths.
- I found a few "wrong" splits for `code` or data without its corresponding `.text` file, so I changed them in `tools/disasm/files.txt`.
- Add the `asm/non_matchings/.../D_80XXXXXX.s` to overlays and code files.
- Update the progress script. 
  - It seems to give the expected percentage with overlays, but `boot` and `code` are wrong.
  - I suspect `boot` is wrong because the spec is not using any C file for it, but I'm not sure
    - Expected: `46712/92464 bytes decompiled in boot 50.51912095518256%`.
    - Current: `-4280/92320 bytes decompiled in boot -4.636048526863085%`
  - I have no idea why `code` is wrong :c
    - Expected: `191136/1069008 bytes decompiled in code 17.879753940101477%`.
    - Current: `117852/1068896 bytes decompiled in code 11.025581534592702%`
  - I made a few filelists to know which files correspond to which category. Those lists are in `tools/filelists/mm.us.rev1/`

Also, with the current spec it is building an OK rom.

### CODE REVIEW

File: src/boot_O2/__osMalloc.c [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 extern OSMesg sArenaLockMsg[1];
 
+#pragma GLOBAL_ASM("asm/non_matchings/boot/__osMalloc/D_80099110.s")
```
Comment (engineer124) [inline src/boot_O2/__osMalloc.c:18]: Here

File: src/boot_O2_g3_trapuv/fault.c [hunk 2]
```diff
@@ -1,9 +1,12 @@
 #include <ultra64.h>
 #include <global.h>
+#include "vt.h"
 
 // TODO move out
 #define OS_CLOCK_RATE 62500000LL
 
+#pragma GLOBAL_ASM("asm/non_matchings/boot/fault/D_800982B0.s")
```
Comment (engineer124) [inline src/boot_O2_g3_trapuv/fault.c:4]: Here

File: src/boot_O2_g3_trapuv/fault_drawer.c [hunk 3]
```diff
@@ -140,6 +143,8 @@ void FaultDrawer_FillScreen() {
 
 #pragma GLOBAL_ASM("asm/non_matchings/boot/fault_drawer/FaultDrawer_FormatStringFunc.s")
 
+#pragma GLOBAL_ASM("asm/non_matchings/boot/fault_drawer/D_80099080.s")
```
Comment (engineer124) [inline src/boot_O2_g3_trapuv/fault_drawer.c:145]: Here

File: src/code/audio_load.c [hunk 4]
```diff
@@ -0,0 +1,162 @@
+#include <ultra64.h>
+#include <global.h>
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/audio_load/func_8018EB60.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/audio_load/func_8018EC4C.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/audio_load/D_801E030C.s")
```
Comment (engineer124) [inline src/code/audio_load.c:7]: Here

File: src/code/code_8019AF00.c [hunk 5]
```diff
@@ -0,0 +1,430 @@
+#include <ultra64.h>
+#include <global.h>
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/D_801E0BD0.s")
```
Comment (engineer124) [inline src/code/code_8019AF00.c:3]: Here

File: src/code/code_8019AF00.c [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019C300.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019C398.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019C5A0.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019C8D8.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/D_801E0C14.s")
```
Comment (engineer124) [inline src/code/code_8019AF00.c:59]: Here

File: src/code/code_801A51F0.c [hunk 7]
```diff
@@ -0,0 +1,33 @@
+#include <ultra64.h>
+#include <global.h>
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A51F0/D_801E0EC0.s")
```
Comment (engineer124) [inline src/code/code_801A51F0.c:3]: Here

File: src/code/code_801AA020.c [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801AA020/func_801AA020.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801AA020/func_801AA248.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801AA020/func_801AA3E4.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801AA020/func_801AA520.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_801AA020/D_801E1100.s")
```
Comment (engineer124) [inline src/code/code_801AA020.c]: Here

File: src/code/flg_set.c [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#include <ultra64.h>
+#include <global.h>
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/func_800B32D0.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/D_801DC120.s")
```
Comment (engineer124) [inline src/code/flg_set.c:5]: Here

File: src/code/flg_set.c [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/func_800B3644.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/D_801DC9B8.s")
```
Comment (engineer124) [inline src/code/flg_set.c:9]: Here

File: src/code/z_kaleido_manager.c [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_801637B4.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_80163804.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_8016388C.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_801638D8.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/D_801DF9C0.s")
```
Comment (engineer124) [inline src/code/z_kaleido_manager.c]: Here

File: src/code/z_play.c [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_8016A168.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_8016A178.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_8016A268.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Play_Init.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/D_801DFA18.s")
```
Comment (engineer124) [inline src/code/z_play.c]: Here

**Review by engineer124 — COMMENTED** [review]
Found some GLOBAL_ASM that don't actually look like functions

**Comment by Thar0** [comment]
🚀 🎊

**Pull request merged**